src/poetry/installation/executor.py

```diff
@@ -88,6 +92,11 @@ def __init__(
         self._shutdown = False
         self._hashes: dict[str, str] = {}
 
+    def set_chef(self, chef: Chef) -> Executor:
+        self._chef = chef
+
+        return self
+
```

Comment:
`set_chef` is only required for tests. It's probably a matter of taste, but I wouldn't introduce a setter just for a test but set the attribute `_chef` directly in the test. I don't think it's too bad to access internals in the unit test of the corresponding module.